forem/forem
# Refactor:Remove ENVied Gem, Use dotenv Instead for Test and Development

## What type of PR is this? (check all applicable)

## Description
ENVied is a gem that is not really necessary for Forem to run. It has the nice feature of "requiring" ENV variables to be present in production but that has caused us more headaches than it seems to be worth. It [used to provide us default values](https://github.com/eval/envied#what-happened-to-default-values) but they have since changed that in the gem in favor of other best practices.

This PR removes the ENVied gem and replaces it with [dotenv](https://github.com/bkeepers/dotenv) for test and development environments. This gem allows us to use a `.env` file to declare all of our environment variables. To begin, there is a rake task that will take a user's `application.yml` file and convert it to a `.env` file. After a couple of weeks I will remove that rake task. 

I also took the opportunity to make our initializers more resilient. Now if Pusher or AWS creds are missing the whole app won't crash, we just simply won't set those values. This will of course likely lead to broken features but it at least gives application users the ability to run the app without them. 

TODO: remove `sample_application.yml` and Figaro gem

## Related Tickets & Documents
https://github.com/orgs/forem/projects/6#card-43004935

## QA Instructions, Screenshots, Recordings
- Pull down the branch
- Run bin/setup
- Run the app and check that a new .env file has been created with everything from your application.yml file. 

## Added tests?

![alt_text](https://media.tenor.com/images/72f0f0be88614674022e91dafd547363/tenor.gif)
